applications/RANSApplication/python_scripts/formulations/utilities.py

```diff
@@ -0,0 +1,244 @@
+
+    if not model.HasModelPart(destination_modelpart_name):
+        model_part = model.CreateModelPart(destination_modelpart_name)
+
+        # TODO: Remove this line once the warnings from connectivity preserve modeller is gone, otherwise,
+        #       output will be cluttered with lots of missing variable warnings
+        RansVariableUtilities.CopyNodalSolutionStepVariablesList(
+            origin_modelpart, model_part)
+
+        # TODO: [PeriodicCondition]
+        #       Currently, all the conditions will be replaced with the given new condition. This is an issue
+        #       in the case of periodic cases in mpi, there we have to put PeriodicConditions in the mdpa file,
+        #       where MetisParitioner will use that condition list to properly partition it. Therefore, "original_condition_name"
+        #       is not used in this method at the moment.
+        #       Following is one of the proposals to make PeriodicConditions to work with connectivity_preserve_modeller.
+        # connectivity_preserve_modeler.GenerateModelPart(
+        #     origin_modelpart, model_part, element_name, condition_name,
+        #     original_condition_name + str(domain_size) + "D" + str(domain_size)
+        #     + "N")
+
+        connectivity_preserve_modeler.GenerateModelPart(
+            origin_modelpart, model_part, element_name, condition_name)
+
+    Kratos.Logger.PrintInfo("RANSModelPartFactory",
+                            "Created " + destination_modelpart_name)
+    return model.GetModelPart(destination_modelpart_name)
+
+
+def CreateResidualBasedBlockBuilderAndSolver(linear_solver, is_periodic,
+                                             communicator):
+    if (is_periodic):
+        return periodic_block_builder_and_solver(linear_solver, communicator)
+    else:
+        return block_builder_and_solver(linear_solver, communicator)
+
+
+def CreateResidualBasedNewtonRaphsonStrategy(model_part, scheme,
+                                             convergence_criteria,
+                                             builder_and_solver,
+                                             max_iterations, compute_reactions,
+                                             reform_dofs_at_each_step,
+                                             move_mesh_flag):
+    return newton_raphson_strategy(model_part, scheme,
+                                   convergence_criteria, builder_and_solver,
+                                   max_iterations, compute_reactions,
+                                   reform_dofs_at_each_step, move_mesh_flag)
+
+
+def CreateLinearSolver(solver_settings):
+    return linear_solver_factory.ConstructSolver(solver_settings)
+
+
+def CreateResidualCriteria(variable_convergence_criteria_info_list):
+    return residual_criteria(variable_convergence_criteria_info_list)
+
+
+def CreateIncrementalUpdateScheme():
+    return incremental_update_static_scheme()
+
+
+def CalculateNormalsOnConditions(model_part):
```

Comment:
I think this should go to the `NormalCalculationUtils`.  Something like `CalculateOnSimplexForFlag`. Then we can call that one in the `ApplySlipProcess` from the base `FluidDynamicsApplication` so you don't need this any more.